miraclelinux/meta-debian-extended
# connman: Update version from 1.36-2.1~deb10u4 to 1.36-2.1~deb10u5

# Purpose of pull request
This version fixes following CVEs.
- CVE-2023-28488

DLA: DLA 3397-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. Build qemuarm64 image
2. Run qemu image
3. Check the connman daemon is working

### Build qemuarm64 image
Add the following to local.conf and build qemuarm64 image.
```
IMAGE_INSTALL_append = " connman"
DISTRO_FEATURES_append = " pam"
```
```
$ bitbake core-image-minimal
```
### Run qemu image
Specify the `slirp` option for enable user networking.
```
$ runqemu qemuarm64 nographic slirp
```
### Check the connman daemon is working
Run following commands to check the connman daemon works.
```
# ps | grep connman
# cat /etc/resolv.conf
# cat /var/log/messages | grep connman
# ip address
# ip route
```
## Test result
### build
Build succeeded.
```
$ bitbake connman
Loading cache: 100% |##########################################################################################################################| Time: 0:00:00
Loaded 2384 entries from dependency cache.
Checking Debian Release ...
Sources.xz is not changed.
Force regenerate source code information.
Parsing Debian Sources.xz ...
Parsing recipes: 100% |########################################################################################################################| Time: 0:00:00
Parsing of 1362 .bb files complete (1360 cached, 2 parsed). 2384 targets, 71 skipped, 6 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.7"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "HEAD:04469d4b27cb5203de34211f3fd8248a510b6c35"
meta-debian-extended = "update-connman_1.36-2.1+deb10u5:9c6ca03c5f943ee6755f07fdadb3c6fa04d419e0"
meta-emlinux         = "HEAD:946bc9498f07741e667ec2b670ad346277d6be0f"
meta-emlinux-private = "HEAD:b290dafea47dfbd8df7ee9b04af38c9b73c87b68"

Initialising tasks: 100% |#####################################################################################################################| Time: 0:00:01
Sstate summary: Wanted 28 Found 2 Missed 26 Current 590 (7% match, 95% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2697 tasks of which 2638 didn't need to be rerun and all succeeded.
```

### Package test
The connman daemon is running when system is started.
```
EMLinux 2.7 qemuarm64 /dev/ttyAMA0

qemuarm64 login: root
root@qemuarm64:~# ps | grep connman
 1656 root     11232 S    /usr/sbin/connmand
 1881 root      3220 S    grep connman
```

The `/etc/resolv.conf` file is created by connman.
```
root@qemuarm64:~# cat /etc/resolv.conf
# Generated by Connection Manager
nameserver 10.0.2.3
nameserver 127.0.0.1
```

According to the `/var/log/messages` log, the connman daemon has setup network interface.
```
root@qemuarm64:~# cat /var/log/messages | grep connman
Apr 28 04:24:03 qemuarm64 daemon.err connmand[1656]: No IPv6 support
Apr 28 04:24:03 qemuarm64 daemon.err connmand[1656]: Failed to open RFKILL control device
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: lo {newlink} index 1 address 00:00:00:00:00:00 mtu 65536
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: lo {newlink} index 1 operstate 0 <UNKNOWN>
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: eth0 {create} index 2 type 1 <ETHER>
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: eth0 {RX} 2 packets 632 bytes
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: eth0 {TX} 1 packets 590 bytes
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: eth0 {update} flags 36866 <DOWN>
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: eth0 {newlink} index 2 address 52:54:00:12:35:02 mtu 1500
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: eth0 {newlink} index 2 operstate 2 <DOWN>
Apr 28 04:24:03 qemuarm64 user.notice kernel: [    8.180196] random: connmand: uninitialized urandom read (8 bytes read)
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: Adding interface eth0 [ ethernet ]
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: eth0 {RX} 2 packets 632 bytes
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: eth0 {TX} 1 packets 590 bytes
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: eth0 {update} flags 102467 <UP,RUNNING,LOWER_UP>
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: eth0 {newlink} index 2 address 52:54:00:12:35:02 mtu 1500
Apr 28 04:24:03 qemuarm64 daemon.info connmand[1656]: eth0 {newlink} index 2 operstate 6 <UP>
Apr 28 04:24:03 qemuarm64 user.notice kernel: [    9.284694] random: connmand: uninitialized urandom read (8 bytes read)
Apr 28 04:24:03 qemuarm64 daemon.warn connmand[1656]: ipconfig state 3 ipconfig method 1
Apr 28 04:24:04 qemuarm64 daemon.info connmand[1656]: eth0 {add} address 10.0.2.15/24 label eth0 family 2
Apr 28 04:24:04 qemuarm64 daemon.info connmand[1656]: eth0 {add} route 10.0.2.0 gw 0.0.0.0 scope 253 <LINK>
Apr 28 04:24:04 qemuarm64 daemon.info connmand[1656]: eth0 {add} route 10.0.2.2 gw 0.0.0.0 scope 253 <LINK>
Apr 28 04:24:04 qemuarm64 daemon.info connmand[1656]: eth0 {add} route 0.0.0.0 gw 10.0.2.2 scope 0 <UNIVERSE>
Apr 28 04:24:04 qemuarm64 daemon.info connmand[1656]: eth0 {add} route 0.0.0.0 gw 10.0.2.2 scope 0 <UNIVERSE>
Apr 28 04:24:04 qemuarm64 daemon.info connmand[1656]: eth0 {add} route 10.0.2.0 gw 0.0.0.0 scope 253 <LINK>
Apr 28 04:24:04 qemuarm64 daemon.info connmand[1656]: eth0 {add} route 10.0.2.2 gw 0.0.0.0 scope 253 <LINK>
Apr 28 04:24:04 qemuarm64 daemon.info dbus-daemon[1649]: [system] Activating service name='org.bluez' requested by ':1.0' (uid=0 pid=1656 comm="/usr/sbin/connmand ") (using servicehelper)
Apr 28 04:24:04 qemuarm64 daemon.info connmand[1656]: eth0 {add} route 82.165.8.211 gw 10.0.2.2 scope 0 <UNIVERSE>
```
Network setting:
```
root@qemuarm64:~# ip address
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP8000> mtu 1500 qdisc pfifo_fast qlen 1000
    link/ether 52:54:00:12:35:02 brd ff:ff:ff:ff:ff:ff
    inet 10.0.2.15/24 brd 10.0.2.255 scope global eth0
       valid_lft forever preferred_lft forever
root@qemuarm64:~# ip route
default via 10.0.2.2 dev eth0 
10.0.2.0/24 dev eth0 scope link  src 10.0.2.15 
10.0.2.2 dev eth0 scope link 
82.165.8.211 via 10.0.2.2 dev eth0 
```
